djstripe/stripe_objects.py

```diff
@@ -954,6 +954,59 @@ def api_retrieve(self, api_key=None):
         return stripe_event
 
 
+class StripePayout(StripeObject):
+    class Meta:
+        abstract = True
+
+    stripe_class = stripe.Payout
+    stripe_dashboard_item_name = "payouts"
+
+    amount = StripeCurrencyField(
+        help_text="Amount to be transferred to your bank account or debit card."
+    )
+    arrival_date = StripeDateTimeField(
+        help_text=(
+            "Date the payout is expected to arrive in the bank. "
+            "This factors in delays like weekends or bank holidays."
+        )
+    )
+    currency = StripeCharField(max_length=3, help_text="Three-letter ISO currency code.")
+    failure_code = StripeCharField(
+        max_length=23,
+        blank=True, null=True,
+        choices=enums.PayoutFailureCode.choices,
+        help_text="Error code explaining reason for transfer failure if available. "
+        "See https://stripe.com/docs/api/python#transfer_failures."
+    )
+    failure_message = StripeTextField(
+        null=True, blank=True,
+        help_text="Message to user further explaining reason for payout failure if available."
+    )
+    method = StripeCharField(
+        max_length=8,
+        choices=enums.PayoutMethod.choices,
+        help_text=(
+            "The method used to send this payout. "
+            "`instant` is only supported for payouts to debit cards."
+        )
+    )
+    statement_descriptor = StripeCharField(
+        max_length=255, null=True, blank=True,
+        help_text="Extra information about a payout to be displayed on the user’s bank statement."
+    )
+    status = StripeCharField(
+        max_length=10,
+        choices=enums.PayoutStatus.choices,
+        help_text=(
+            "Current status of the payout. "
+            "A payout will be `pending` until it is submitted to the bank, at which point it "
+            "becomes `in_transit`. I t will then change to paid if the transaction goes through. "
+            "If it does not go through successfully, its status will change to `failed` or `canceled`."
+        )
+    )
+    type = StripeCharField(max_length=12, choices=enums.PayoutType.choices)
```

Comment:
Shadowing function names as class attributes is fine because they're scoped to the class definition itself.

This would break:

```py
type = StripeCharField(max_length=12, choices=enums.PayoutType.choices)
if isinstance(foo, type(bar)):
    raise oops
```